### PR TITLE
Feature/core update

### PR DESCRIPTION
**Pull request by jasoncavett**
Updating to build against `netstandard1.6` and work with NET Core. Upgraded references to Nito.AsyncEx.\* and adjust the rest of the build and related code.

### CODE REVIEW

File: Shields.DataStructures.Async/AsyncBoundedQueue.cs [hunk 1]
```diff
@@ -299,40 +292,38 @@ public Task EnqueueAsync(T value, CancellationToken cancellationToken)
...
+            enqueueQueue.DequeueAll();
         }
 
         /// <summary>
-        /// Returns a disposable that cancels all waiting enqueue operations.
+        /// Cancels all waiting enqueue operations.
         /// </summary>
-        /// <returns>The disposable that cancels all waiting enqueue operations.</returns>
-        public IDisposable CancelAllEnqueue()
+        /// <param name="cancellationToken"></param>
```
Comment (timothy-shields) [inline Shields.DataStructures.Async/AsyncBoundedQueue.cs]: Please change the parameter comment to the following.

```
/// <param name="cancellationToken">The cancellation token.</param>
```

File: Shields.DataStructures.Async/AsyncQueue.cs [hunk 2]
```diff
@@ -144,28 +143,27 @@ public void Enqueue(T value)
...
+            dequeueQueue.DequeueAll(value);
         }
         
         /// <summary>
-        /// Returns a disposable that cancels all waiting DequeueAsync calls.
+        /// Cancels all waiting DequeueAsync calls.
         /// </summary>
-        /// <returns>The disposable that cancels all waiting DequeueAsync calls.</returns>
-        public IDisposable CancelAllDequeue()
+        /// <param name="cancellationToken"></param>
```
Comment (timothy-shields) [inline Shields.DataStructures.Async/AsyncQueue.cs]: Please change the parameter comment to the following.

```
/// <param name="cancellationToken">The cancellation token.</param>
```

File: Shields.DataStructures.Async/AsyncStack.cs [hunk 3]
```diff
@@ -144,28 +143,27 @@ public void Push(T value)
...
+            popQueue.DequeueAll(value);
         }
 
         /// <summary>
-        /// Returns a disposable that cancels all waiting PopAsync calls.
+        /// Cancels all waiting PopAsync calls.
         /// </summary>
-        /// <returns>The disposable that cancels all waiting PopAsync calls.</returns>
-        public IDisposable CancelAllPop()
+        /// <param name="cancellationToken"></param>
```
Comment (timothy-shields) [inline Shields.DataStructures.Async/AsyncStack.cs]: Please change the parameter comment to the following.

```
/// <param name="cancellationToken">The cancellation token.</param>
```

**Comment by timothy-shields** [comment]
You can completely remove `AsyncReaderWriterLockDictionary.UpgradeableReaderKey`, it isn't referenced anymore. I also left 3 line comments about how to document `CancellationToken` parameters.

**Comment by timothy-shields** [comment]
It looks like the unit tests didn't actually run:

https://ci.appveyor.com/project/timothy-shields/async-data-structures/build/1.0.6

**Comment by jasoncavett** [comment]
The checks are going to fail because AppVeyor does not currently provide a build profile for MSBuild 15.

**Pull request merged**